test/e2e/kcp_upgrade.go

```diff
@@ -71,6 +69,8 @@ func KCPUpgradeSpec(ctx context.Context, inputGetter func() KCPUpgradeSpecInput)
 	It("Should successfully upgrade Kubernetes, DNS, kube-proxy, and etcd", func() {
 
 		By("Creating a workload cluster")
+		Expect(input.E2EConfig.Variables).To(HaveKey(clusterctl.KubernetesVersion))
+		Expect(input.E2EConfig.Variables).To(HaveKey(clusterctl.CNIPath))
```

Comment:
Do we need these checks for every test? We already throw error if they are empty during validation.

Reply:
I don't think we should enforce variables to exist during validation... I missed this.
Each spec might require different variables

Reply:
KubernetesVersion and CNI should exist for all tests? If not we should remove those checks from validation.